src/http/headers.cr

```diff
@@ -356,4 +377,10 @@ struct HTTP::Headers
       end
     end
   end
+
+  private def check_writeable
+    if read_only?
+      raise ReadOnlyError.new("HTTP::Headers are in read-only mode (it probably means the HTTP::Server headers were already sent)")
```

Comment:
This has nothing to do with `HTTP::Server`. You can build and send headers without a server.
I would also suggest to let the error message explicitly say that headers were already sent. Maybe even make `read_only!` nodoc and rename it to `headers_written`. I don't think there's a need to call this method directly, read-only mode is not a standalone concept. We should assume read-only mode is only an implicity result of writing the headers to an IO.

```suggestion
      raise ReadOnlyError.new("HTTP::Headers were already sent)")
```

Reply:
Good idea